npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater_spec.rb

```diff
@@ -197,202 +142,258 @@
+                "https://github.com/hmarr/dependabot-test-private-npm-package.git/"
               ]
             )
         end
       end
     end
+  end
 
-    context "because we're updating to a nonexistent version" do
-      let(:files) { project_dependency_files("npm6/simple") }
+  describe "npm 7 specific" do
+    # NOTE: This used to raise in npm 6
+    context "when there is a private git dep we don't have access to" do
+      let(:files) { project_dependency_files("npm7/github_dependency_private") }
 
-      let(:dependency_name) { "fetch-factory" }
-      let(:version) { "5.0.2" }
-      let(:requirements) do
-        [{
-          file: "package.json",
-          requirement: "^5.0.2",
-          groups: ["dependencies"],
-          source: nil
-        }]
-      end
+      let(:dependency_name) { "strict-uri-encode" }
+      let(:version) { "1.1.0" }
+      let(:requirements) { [] }
 
-      it "raises an unhandled error" do
-        expect { updated_npm_lock_content }.
-          to raise_error(Dependabot::InconsistentRegistryResponse)
+      it "updates the dependency and leaves the private git dep alone" do
+        parsed_lockfile = JSON.parse(updated_npm_lock_content)
+        expect(parsed_lockfile.fetch("dependencies")["strict-uri-encode"]["version"]).
+          to eq("1.1.0")
+        expect(parsed_lockfile.fetch("dependencies")["bus-replacement-service"]["version"]).
+          to include("19c4dba3bfce7574e28f1df2138d47ab4cc665b3")
       end
     end
+  end
 
-    context "with a dependency that can't be found" do
-      let(:files) { project_dependency_files("npm6/nonexistent_dependency_yanked_version") }
+  %w(npm6 npm7).each do |npm_version|
+    describe "#{npm_version} updates" do
+      let(:files) { project_dependency_files("#{npm_version}/simple") }
 
-      it "raises a helpful error" do
-        expect { updated_npm_lock_content }.
-          to raise_error(Dependabot::PrivateSourceAuthenticationFailure)
+      it "has details of the updated item" do
+        parsed_lockfile = JSON.parse(updated_npm_lock_content)
+        expect(parsed_lockfile["dependencies"]["fetch-factory"]["version"]).
+          to eq("0.0.2")
       end
-    end
 
-    context "with a git reference that Yarn would find but npm wouldn't" do
-      let(:files) { project_dependency_files("npm6/git_dependency_yarn_ref") }
+      context "when the requirement has not been updated" do
+        let(:requirements) { previous_requirements }
 
-      it "raises a helpful error" do
-        expect { updated_npm_lock_content }.
-          to raise_error(Dependabot::DependencyFileNotResolvable)
-      end
-    end
+        it "has details of the updated item" do
+          parsed_lockfile = JSON.parse(updated_npm_lock_content)
+          expect(parsed_lockfile["dependencies"]["fetch-factory"]["version"]).
+            to eq("0.0.2")
 
-    context "with a corrupted npm lockfile (version missing)" do
```

Comment:
No longer failing in npm 7, npm 6 spec is here https://github.com/dependabot/dependabot-core/blob/bfbc137cddb2da65f0e37f6ee4fae8c6dee75067/npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater_spec.rb#L64